tests/datasets/test__init__.py

```diff
@@ -0,0 +1,424 @@
+# limitations under the License.
+
+
+import pytest
+
+from faculty import datasets
+
+
+@pytest.fixture
+def mock_client(mocker):
+    session = mocker.Mock()
+    get_session_mock = mocker.patch(
+        "faculty.datasets.get_session", return_value=session
+    )
+
+    project_id = mocker.Mock()
+
+    object_client = mocker.Mock()
+    mocker.patch("faculty.datasets.ObjectClient", return_value=object_client)
+
+    yield object_client, project_id
+
+    get_session_mock.assert_called_once_with()
+
+
+def test_datasets_ls_all_files(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    list_response = mocker.Mock()
+    mock_object = mocker.Mock()
+    mock_object.path = "test-path"
+    list_response.objects = [mock_object]
+    list_response.next_page_token = None
+    object_client.list.return_value = list_response
+
+    objects = datasets.ls("test-prefix", project_id, show_hidden=True)
+    assert objects == ["test-path"]
+    object_client.list.assert_called_once_with(project_id, "test-prefix")
+
+
+def test_datasets_ls_non_hidden_files(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    list_response = mocker.Mock()
+    mock_object = mocker.Mock()
+    mock_object.path = ".test-hidden-path"
+    list_response.objects = [mock_object]
+    list_response.next_page_token = None
+    object_client.list.return_value = list_response
+
+    objects = datasets.ls("test-prefix", project_id)
+    assert objects == []
+    object_client.list.assert_called_once_with(project_id, "test-prefix")
+
+
+def test_datasets_ls_with_continuation(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    list_response1 = mocker.Mock()
+    mock_object1 = mocker.Mock()
+    mock_object1.path = ".test-hidden-path"
+    list_response1.objects = [mock_object1]
+    list_response1.next_page_token = mocker.Mock()
+
+    list_response2 = mocker.Mock()
+    mock_object2 = mocker.Mock()
+    mock_object2.path = "test-path"
+    list_response2.objects = [mock_object2]
+    list_response2.next_page_token = None
+
+    object_client.list.side_effect = [list_response1, list_response2]
+
+    objects = datasets.ls("test-prefix", project_id)
+    assert objects == ["test-path"]
+    object_client.list.assert_has_calls(
+        [
+            mocker.call(project_id, "test-prefix"),
+            mocker.call(
+                project_id, "test-prefix", list_response1.next_page_token
+            ),
+        ]
+    )
+
+
+def test_datasets_glob_path(mocker):
+    project_id = mocker.Mock()
+
+    content_mock = mocker.Mock()
+    ls_mock = mocker.patch("faculty.datasets.ls", return_value=content_mock)
+    fnmatch_filter_mock = mocker.patch(
+        "fnmatch.filter", return_value=mocker.Mock()
+    )
+
+    pattern_mock = mocker.Mock()
+    datasets.glob(
+        pattern_mock,
+        prefix="project-path",
+        project_id=project_id,
+        show_hidden=True,
+        object_client=None,
+    )
```

Comment:
Want to assert that this returns the return value of `fnmatch.filter`?

I'd also be happy not mocking `fnmatch.filter` in this case.